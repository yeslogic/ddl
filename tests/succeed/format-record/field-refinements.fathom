let format = {
    magic <- u64le where u64_eq magic 0x00ffffffffffff00,
    len <- u8 where u8_lte len 16,
    data <- repeat_len64 (u8_extend_u64 len) u8,
};

let _ : Repr format -> { magic : U64, len : U8, data : Array64 (u8_extend_u64 len) U8 } =
    fun x => x;

{}
