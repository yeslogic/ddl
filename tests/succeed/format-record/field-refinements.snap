stdout = '''
let format : Format = {
    magic <- u64le where magic == (0xffffffffffff00 : U64),
    len <- u8 where len <= (16 : U8),
    data <- array8 len u8,
};
let _ : { magic : U64, len : U8, data : Array8 len U8 } -> {
    magic : U64,
    len : U8,
    data : Array8 len U8,
} = fun x => x;
{} : {}
'''
stderr = ''
